domains/certificates/Query.API/API/ContractService/ContractServiceImpl.cs

```diff
@@ -42,10 +42,12 @@ public ContractServiceImpl(IMeteringPointsClient client, ICertificateIssuingCont
 
         var documents = await repository.GetByGsrn(gsrn, cancellationToken);
 
-        var documentExist = CheckPeriod(documents, startDate, endDate);
-        if (documentExist != null)
+        foreach (var document in documents)
         {
-            return documentExist;
+            if (!CanCreateContract(document, startDate, endDate))
```

Comment:
A minor if you rename it to `CannotCreateContract` then we could avoid the `!`which is easy to miss when skimming the code

But just from reading the name i have no idea what business check that prevents the contract from being created.

Reply:
I don't agree. The function checks whether or not a contract can get created. It also makes sure there isn't any overlapping periods between contracts.
So i don't agree, that the function should be renamed

Reply:
But why not rename it to `Cannot`? The only place you are using it is when it cannot create

Reply:
Because the return statement in the functions returns true if the contract is allowed to be created. False if it cannot. Its a oneliner, which potentially can be used in the future to make this check.
If i renamed it, it would just like like this:
`!CannotCreateContract`